sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/KotlinSensor.kt

```diff
@@ -127,6 +127,27 @@ class KotlinSensor(
         sensorDuration?.stop()
     }
 
+    private fun reuseCPDTokens(inputFile: InputFile,sensorContext: SensorContext): Boolean {
+        if (!sensorContext.isCacheEnabled) {
+            return false
+        }
+        val cpdTokensCache = CPDTokensCache(sensorContext.previousCache(), sensorContext.nextCache())
+        val previousTokens: List<CPDToken>? = cpdTokensCache.loadFromPreviousAnalysis(inputFile)
+        return if (previousTokens != null) {
+            val tokensToReuse = sensorContext.newCpdTokens().onFile(inputFile)
+            previousTokens.forEach { tokensToReuse.addToken(it.range, it.text) }
+            tokensToReuse.save()
+            try {
+                cpdTokensCache.copyForNextAnalysis(inputFile)
+                true
+            } catch (_: IllegalArgumentException) {
+                false
+            }
+        } else {
+            false
+        }
```

Comment:
This construct of checking for `null` and acting correspondingly is a typical candidate for a `?.let` construction in Kotlin. The advantage is that you save some boilerplate code and the semantic dependencies in the code are more explicit.